www/experiments.php

```diff
@@ -554,99 +554,170 @@ function refreshExperimentFormState(){
             echo "$id";
                              } ?>';
         if(priorState && currentTestID && priorState.indexOf("resubmit="+ currentTestID) > -1 ){
-            var form = $("form.experiments_grades");
-            form[0].reset();
+            var form = document.querySelector("form.experiments_grades");
+            form.reset();
             var pairs = priorState.split("&");
 
                 pairs.forEach(pair => {
                     var keyval = pair.split("=");
+                    if( keyval[0]){
+                        keyval[0] = decodeURIComponent(keyval[0]);
+                    }
+                    if( keyval[1]){
+                        keyval[1] = decodeURIComponent(keyval[1]);
+                    }
```

Comment:
alternatively a shorter expression could be
```js
let keyval = pair.split("=").map(decodeURIComponent);
```

Reply:
elegant!